data_adapter_oemof/build_datapackage.py

```diff
@@ -91,8 +91,38 @@ def __split_timeseries_into_years(parametrized_sequences):
 
         return split_dataframes
 
-    def get_foreign_keys(self):
-        pass
+    @staticmethod
+    def get_foreign_keys(
+        process_busses: list, mapper: Mapper, components: list
+    ) -> list:
+        """
+        Writes Foreign keys for one process.
+        Searches in adapter class for sequences fields
+        :param components:
+        :rtype: list
+        :param process_busses:
+        :param adapter:
+        :return:
+        """
+        new_foreign_keys = []
+        components = pd.DataFrame(components)
+        for bus in process_busses:
+            new_foreign_keys.append(
+                {"fields": bus, "reference": {"fields": "name", "resource": "bus"}}
+            )
+
+        for field in dataclasses.fields(mapper.adapter):
+            if mapper.is_sequence(field.type):
+                if all(components[field.name].isin(mapper.timeseries.columns)):
```

Comment:
This does not solve the problem. First, we have to map the field name and afterwards we have to check whether mapped key can be found in data. If both is fulfilled this key is a candidate for a sequence. **then** we can check if mapped key `is_sequence`. I can implement this part

Reply:
As far as I know the scalars are already mapped and the timeseries column names have to fit to the scalars within the components[field.name] already so we dont have to map again.

Reply:
Maybe you are right - have to check it in debug mode.

Reply:
Did you check/resolve this?